mypy/checkpattern.py

```diff
@@ -553,6 +554,15 @@ def visit_class_pattern(self, o: ClassPattern) -> PatternType:
             and isinstance(get_proper_type(type_info.type), AnyType)
         ):
             typ = type_info.type
+        elif (
+            isinstance(type_info, Var)
+            and type_info.type is not None
```

Comment:
these two conditions are also checked in the elif before and below, but I did not bother to try to optimize them away.